api/models/node.py

```diff
@@ -25,3 +25,6 @@ def __eq__(self, __value: object) -> bool:
 
     def __hash__(self) -> int:
         return hash(self.id)
+
+    def __contains__(self, item):
+        return any(str(item).lower in (self.id.lower(), str(d).lower, str(self.data[d])) for d in self.data.keys())
```

Comment:
I think search shouldn't be in the `Node` class, but since search is not the point of this PR, it's not really that big of an issue. It should probably be removed from here when search is implemented.

Reply:
this is just the override of the node's 'in' method, it is a general node feature which will implicitly make search easier